### PR TITLE
zio integration

### PR DESCRIPTION
**Pull request by dieproht**
Hey Chris, 
with this PR I made an integration for zio like there already exists for cats-effect, scalaz, try/future. Maybe you can have a look?
Karl

### CODE REVIEW

File: modules/zio/src/main/scala/scalacache/zio.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package scalacache
+
+import _root_.zio.Task
+
+import scala.util.control.NonFatal
+
+object zio {
```
Comment (dpennell) [inline modules/zio/src/main/scala/scalacache/zio.scala]: Can we have this as `ZioEffect` similar to `scalacache.CatsEffect`.
It is common to `import scalacache._` and this ends up shading the `zio` namespace.

Comment (dieproht) [reply modules/zio/src/main/scala/scalacache/zio.scala]: Thanks for your feedback. That sounds reasonable. I renamed the module object according your suggestion.

**Comment by dpennell** [comment]
Thanks for contributing this!

**Comment by lewisjkl** [comment]
Thanks for the contribution here! I am going to close this since the library as a whole is moving away from having different modes (see #345).

**Pull request closed without merging**